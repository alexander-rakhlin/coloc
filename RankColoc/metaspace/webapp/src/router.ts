import Vue from 'vue';
import VueRouter from 'vue-router';
import AboutPage from './modules/App/AboutPage.vue';
import DatasetsPage from './modules/Datasets/DatasetsPage.vue';
import {DialogPage, ResetPasswordPage} from './modules/Account';

Vue.use(VueRouter);

const asyncPages = {
  AnnotationsPage: () => import(/* webpackPrefetch: true, webpackChunkName: "AnnotationsPage" */ './modules/Annotations/AnnotationsPage.vue'),
  DatasetSummary: () => import(/* webpackPrefetch: true, webpackChunkName: "DatasetSummary" */ './modules/Datasets/summary/DatasetSummary.vue'),
  MetadataEditPage: () => import(/* webpackPrefetch: true, webpackChunkName: "MetadataEditPage" */ './modules/MetadataEditor/MetadataEditPage.vue'),
  ImageAlignmentPage: () => import(/* webpackPrefetch: true, webpackChunkName: "ImageAlignmentPage" */ './modules/ImageAlignment/ImageAlignmentPage.vue'),
  UploadPage: () => import(/* webpackPrefetch: true, webpackChunkName: "UploadPage" */ './modules/MetadataEditor/UploadPage.vue'),

  // These pages are relatively small as they don't have any big 3rd party dependencies, so pack them together
  DatasetTable: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Datasets/list/DatasetTable.vue'),
  HelpPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/App/HelpPage.vue'),
  EditUserPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/UserProfile/EditUserPage.vue'),
  CreateGroupPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/GroupProfile/CreateGroupPage.vue'),
  ViewGroupPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/GroupProfile/ViewGroupPage.vue'),
  ViewProjectPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Project/ViewProjectPage.vue'),
  ProjectsListPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Project/ProjectsListPage.vue'),
};

const convertLegacyHashUrls = () => {
  const {pathname, hash} = window.location;
  if (pathname === '/' && hash && hash.startsWith('#/')) {
    history.replaceState(undefined, undefined, hash.slice(1));
  }
};
convertLegacyHashUrls();

const router = new VueRouter({
  mode: 'history',
  routes: [
    { path: '/', redirect: '/manualsort' },
    { path: '/about', component: AboutPage },
    { path: '/annotations', component: asyncPages.AnnotationsPage },
    {
      path: '/datasets',
      component: DatasetsPage,
      children: [
        {path: '', component: asyncPages.DatasetTable},
        {path: 'summary', component: asyncPages.DatasetSummary},
      ]
    },
    { path: '/datasets/edit/:dataset_id', name: 'edit-metadata', component: asyncPages.MetadataEditPage },
    { path: '/datasets/:dataset_id/add-optical-image', name: 'add-optical-image', component: asyncPages.ImageAlignmentPage },
    { path: '/upload', component: asyncPages.UploadPage },
    { path: '/help', component: asyncPages.HelpPage },
    { path: '/user/me', component: asyncPages.EditUserPage },

    { path: '/admin/health', component: async () => await import('./modules/Admin/SystemHealthPage.vue') },
    { path: '/admin/groups', component: async () => await import('./modules/Admin/GroupsListPage.vue') },

    { path: '/account/sign-in', component: DialogPage, props: {dialog: 'signIn'} },
    { path: '/account/create-account', component: DialogPage, props: {dialog: 'createAccount'} },
    { path: '/account/forgot-password', component: DialogPage, props: {dialog: 'forgotPassword'} },
    { path: '/account/reset-password', component: ResetPasswordPage },

    { path: '/group/create', component: asyncPages.CreateGroupPage },
    { path: '/group/:groupIdOrSlug', name: 'group', component: asyncPages.ViewGroupPage },
    { path: '/project/:projectIdOrSlug', name: 'project', component: asyncPages.ViewProjectPage },
    { path: '/projects', component: asyncPages.ProjectsListPage },
    { path: '/manualsort', component: async () => (await import(/* webpackPrefetch: true, webpackChunkName: "ImageClassifierPage" */ './modules/ImageClassifier')).ImageClassifierPage  },
  ],
  parseQuery: query => {
    var res: Record<string, string> = {};

    query = query.trim().replace(/^(\?|#|&)/, '');

    if (!query) {
      return res
    }

    query.split('&').forEach(function (param) {
      var parts = param.split('=');
      var key = decodeURIComponent(parts.shift()!);
      var val = parts.length > 0
        ? decodeURIComponent(parts.join('='))
        : null;

      if (res[key] === undefined) {
        res[key] = val!;
      } else if (Array.isArray(res[key])) {
        (res[key] as any).push(val);
      } else {
        (res[key] as any) = [res[key], val];
      }
    });

    return res
  },
  stringifyQuery: (obj: object) => {
    var res = obj ? Object.keys(obj).map(function (key) {
      var val = (obj as Record<string,string>)[key];

      if (val === undefined) {
        return ''
      }

      if (val === null) {
        return encode(key)
      }

      if (Array.isArray(val)) {
        var result: string[] = [];
        val.forEach(function (val2) {
          if (val2 === undefined) {
            return
          }
          if (val2 === null) {
            result.push(encode(key));
          } else {
            result.push(encode(key) + '=' + encode(val2));
          }
        });
        return result.join('&')
      }

      return encode(key) + '=' + encode(val)
    }).filter(function (x) { return x.length > 0; }).join('&') : null;
    return res ? ("?" + res) : ''
  },

});

function encode(str: string) {
  let newStr = '';
  for(let i = 0; i < (str||'').length; i++) {
    if ('+ ,.@:;'.includes(str[i])) {
      newStr += str[i];
    } else {
      newStr += encodeURIComponent(str[i]);
    }
  }
  return newStr;
}

export default router;
