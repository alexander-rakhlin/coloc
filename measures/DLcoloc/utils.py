import pandas as pd
import numpy as np


def train_test_split(df, test_fold=5, n_folds=5):
    assert 1 <= test_fold <= n_folds
    datasets = sorted(df.groupby('datasetId').groups)
    datasets_count = len(datasets)
    fold_length = int(np.ceil(datasets_count / n_folds))
    test_index = range(fold_length * (test_fold - 1), min(datasets_count, fold_length * test_fold))
    train_index = [i for i in range(datasets_count) if i not in test_index]
    test_datasets = [datasets[i] for i in test_index]
    train_datasets = [datasets[i] for i in train_index]
    train_df = df.loc[df['datasetId'].isin(train_datasets)]
    test_df = df.loc[df['datasetId'].isin(test_datasets)]
    assert len(train_df) + len(test_df) == len(df)
    assert len(pd.merge(train_df, test_df, on='datasetId', how='inner')) == 0
    return train_df, test_df


if __name__ == '__main__':
    data_df = pd.read_csv('Data/coloc_gs.csv')
    for test_fold in range(1, 6):
        train_df, test_df = train_test_split(data_df, test_fold=test_fold, n_folds=5)
        print(len(test_df))

