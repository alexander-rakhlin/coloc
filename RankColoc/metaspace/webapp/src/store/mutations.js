import {isEqual, omit, pull, without} from 'lodash-es';
import router from '../router';
import compare from '../lib/compare';

import {
  decodeParams,
  encodeParams,
  encodeSections,
  encodeSortOrder,
  FILTER_SPECIFICATIONS,
  getFilterInitialValue,
  stripFilteringParams,
} from '../modules/Filters';
import {DEFAULT_ANNOTATION_VIEW_SECTIONS, DEFAULT_COLORMAP, DEFAULT_TABLE_ORDER} from '../modules/Filters/url';


function updatedLocation(state, filter) {
  let query = encodeParams(filter, state.route.path, state.filterLists);

  state.lastUsedFilters[state.route.path] = {
    filter,
    query,
    order: state.orderedActiveFilters
  };

  return {
    query: Object.assign(query, stripFilteringParams(state.route.query))
  };
}

function replaceURL(state, filter) {
  router.replace(updatedLocation(state, filter));
}

function pushURL(state, filter) {
  router.push(updatedLocation(state, filter));
}

function sortFilterKeys(keys) {
  keys.sort((a, b) => compare(FILTER_SPECIFICATIONS[a].sortOrder || 100, FILTER_SPECIFICATIONS[b].sortOrder || 100));
}

function updateFilter(state, filter, routerAction = null) {
  let active = [];

  // drop unset filters
  for (var i = 0; i < state.orderedActiveFilters.length; i++) {
    let key = state.orderedActiveFilters[i];
    if (filter[key] !== undefined)
      active.push(key);
  }

  // append newly added filters to the end
  for (var key in filter)
    if (filter[key] !== undefined &&
      active.indexOf(key) == -1)
      active.push(key);

  // sort
  sortFilterKeys(active);

  const changedFilterSet = !isEqual(state.orderedActiveFilters, active);

  state.orderedActiveFilters = active;
  if (routerAction === 'push' || (routerAction == null && changedFilterSet))
    pushURL(state, filter);
  else
    replaceURL(state, filter);
}

export default {
  updateFilter (state, filter) {
    updateFilter(state, filter);
  },

  replaceFilter (state, filter) {
    updateFilter(state, filter, 'replace');
  },

  addFilter (state, name) {
    const initialValue = getFilterInitialValue(name, state.filterLists);

    // FIXME: is there any way to access getters here?
    let filter = Object.assign(decodeParams(state.route, state.filterLists),
                               {[name]: initialValue});

    state.orderedActiveFilters.push(name);
    pushURL(state, filter);
  },

  updateFilterOnNavigate(state, to) {
    const newActiveFilters = Object.keys(decodeParams(to, state.filterLists));

    const removed = without(state.orderedActiveFilters, ...newActiveFilters);
    const added = without(newActiveFilters, ...state.orderedActiveFilters);

    pull(state.orderedActiveFilters, ...removed);
    state.orderedActiveFilters.push(...added);
    sortFilterKeys(state.orderedActiveFilters);
  },

  setFilterListsLoading(state) {
    state.filterListsLoading = true;
  },

  setFilterLists(state, filterLists) {
    state.filterLists = filterLists;
    state.filterListsLoading = false;
  },

  setAnnotation(state, annotation) {
    state.annotation = annotation;
  },

  updateAnnotationTableStatus(state, isLoading) {
    state.tableIsLoading = isLoading;
  },

  startTour(state, tourDescription) {
    state.currentTour = tourDescription;
  },

  endTour(state) {
    state.currentTour = null;
  },

  updateAnnotationViewSections(state, activeSections) {
    const sections = encodeSections(activeSections);
    const defaultSections = encodeSections(DEFAULT_ANNOTATION_VIEW_SECTIONS);
    router.replace({
      query: sections !== defaultSections
        ? { ...state.route.query, sections }
        : omit(state.route.query, 'sections'),
    });
  },

  setColormap(state, cmap) {
    router.replace({
      query: cmap !== DEFAULT_COLORMAP
        ? {...state.route.query, cmap}
        : omit(state.route.query, 'cmap'),
    });
  },

  setCurrentPage(state, page) {
    router.replace({
      query: page !== 1
        ? { ...state.route.query, page }
        : omit(state.route.query, 'page'),
    });
  },

  setSortOrder(state, sortOrder) {
    const sort = encodeSortOrder(sortOrder);
    const defaultSort = encodeSortOrder(DEFAULT_TABLE_ORDER);
    router.replace({
      query: sort !== defaultSort
        ? { ...state.route.query, sort }
        : omit(state.route.query, 'sort'),
    });
  },

  setDatasetTab(state, tab) {
    const path = {
      'list': '/datasets',
      'summary': '/datasets/summary'
    }[tab.toLowerCase()];
    router.replace({
      path,
      query: state.route.query,
    });
  }
};
