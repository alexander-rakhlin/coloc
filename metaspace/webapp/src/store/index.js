import Vue from 'vue';
import Vuex from 'vuex';
Vue.use(Vuex);

import getters from './getters.js';
import mutations from './mutations.js';
import actions from './actions.js';
import {accountModule} from '../modules/Account';

const store = new Vuex.Store({
  state: {
    // names of currently shown filters
    orderedActiveFilters: [],

    filterLists: null,
    filterListsLoading: false,

    // currently selected annotation
    annotation: undefined,

    // is annotation table loading?
    tableIsLoading: true,

    lastUsedFilters: {},

    currentTour: null,
  },

  getters,
  mutations,
  actions,
  modules: {
    account: accountModule
  }
})

export default store;
