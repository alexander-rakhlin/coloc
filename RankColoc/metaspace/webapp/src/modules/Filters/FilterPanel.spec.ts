import { mount } from '@vue/test-utils';
import Vue from 'vue';
import FilterPanel from './FilterPanel.vue';
import router from '../../router';
import { initMockGraphqlClient, provide } from '../../../tests/utils/mockGraphqlClient';
import Vuex from 'vuex';
import store from '../../store/index';
import { sync } from 'vuex-router-sync';
import { encodeParams } from './url';


Vue.use(Vuex);
sync(store, router);

describe('FilterPanel', () => {
  const allFilters = {
    database: 'HMDBv4',
    group: '0123',
    project: '4567',
    submitter: '89AB',
    datasetIds: ['CDEF','GHIJ'],
    minMSM: 0.5,
    compoundName: 'C8H20NO6P',
    adduct: '+K',
    mz: "296.0659",
    fdrLevel: 0.1,
    polarity: 'Positive',
    organism: 'cow',
    organismPart: 'stomach',
    condition: 'hungry',
    growthConditions: 'paddock',
    ionisationSource: 'MALDI',
    maldiMatrix: 'DHB',
    analyzerType: 'FTICR',
    simpleQuery: 'foo',
    metadataType: 'ims'
  };

  beforeEach(() => {
    initMockGraphqlClient();
  });

  it('should match snapshot (no filters)', async () => {
    router.replace({ path: '/annotations' });
    store.commit('updateFilter', {});
    const propsData = { level: 'annotation' };
    const wrapper = mount(FilterPanel, { router, provide, store, propsData, sync: false });
    await Vue.nextTick();

    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot (all annotation filters)', async () => {
    router.replace({ path: '/annotations' });
    store.commit('updateFilter', allFilters);
    const propsData = { level: 'annotation' };
    const wrapper = mount(FilterPanel, { router, provide, store, propsData, sync: false });
    await Vue.nextTick();

    expect(wrapper).toMatchSnapshot();
  });

  it('should update the route when filters change', async () => {
    router.replace({ path: '/annotations' });
    store.commit('updateFilter', allFilters);
    const propsData = { level: 'annotation' };
    const wrapper = mount(FilterPanel, { router, provide, store, propsData, sync: false });
    const newFilters = {
      simpleQuery: 'lorem ipsum',
      database: 'CHEBI',
      project: 'abc',
      datasetIds: ['aaa','bbb'],
      minMSM: "0.1",
      mz: "296.1"
    };
    await Vue.nextTick();

    // simpleQuery - SearchBox
    wrapper.find('[data-test-key="simpleQuery"] input').setValue(newFilters.simpleQuery);
    // database - SingleSelectFilter
    wrapper.find('[data-test-key="database"] mock-el-select').vm.$emit('change', newFilters.database);
    // project - SearchableFilter [multiple=false]
    wrapper.find('[data-test-key="project"] mock-el-select').vm.$emit('change', newFilters.project);
    // datasetIds - SearchableFilter [multiple=true]
    wrapper.find('[data-test-key="datasetIds"] mock-el-select').vm.$emit('change', newFilters.datasetIds);
    // minMSM - InputFilter
    wrapper.find('[data-test-key="minMSM"] .tf-value-span').trigger('click');
    await Vue.nextTick();
    wrapper.find('[data-test-key="minMSM"] input').setValue(newFilters.minMSM);
    // mz - MzFilter
    wrapper.find('[data-test-key="mz"] .tf-value-span').trigger('click');
    await Vue.nextTick();
    wrapper.find('[data-test-key="mz"] input').setValue(newFilters.mz);
    await Vue.nextTick();

    expect(router.currentRoute.query).toEqual(expect.objectContaining(encodeParams(newFilters)));
  });

  it('should be able to add a filter', async () => {
    router.replace({ path: '/annotations' });
    store.commit('updateFilter', {});
    const propsData = { level: 'annotation' };
    const wrapper = mount(FilterPanel, { router, provide, store, propsData, sync: false });
    await Vue.nextTick();
    expect(wrapper.find('[data-test-key="project"]').exists()).toEqual(false);

    wrapper.find('mock-el-select').vm.$emit('change', 'project');
    await Vue.nextTick();

    expect(wrapper.find('[data-test-key="project"]').exists()).toEqual(true);
  });

  it('should be able to remove a filter', async () => {
    router.replace({ path: '/annotations' });
    store.commit('updateFilter', allFilters);
    const propsData = { level: 'annotation' };
    const wrapper = mount(FilterPanel, { router, provide, store, propsData, sync: false });
    await Vue.nextTick();
    expect(wrapper.find('[data-test-key="project"]').exists()).toEqual(true);

    wrapper.find('[data-test-key="project"] .tf-remove').trigger('click');
    await Vue.nextTick();

    expect(wrapper.find('[data-test-key="project"]').exists()).toEqual(false);
  });
});
