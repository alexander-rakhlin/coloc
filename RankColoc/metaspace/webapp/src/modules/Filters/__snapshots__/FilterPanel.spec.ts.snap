// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterPanel should match snapshot (all annotation filters) 1`] = `
<div class="filter-panel">
  <!---->
  <div class="tf-outer" data-test-key="simpleQuery" filterkey="simpleQuery" name="Simple query" options="">
    <div class="tf-name">
      <i class="el-icon-search" style="color: white;"></i>
    </div>
    <input type="text" placeholder="enter any keywords" class="tf-value tf-value-input">
    <!---->
  </div>
  <div class="tf-outer" data-test-key="polarity" filterkey="polarity">
    <div class="tf-name">
      Polarity:
    </div>
    <mock-el-popover trigger="click" placement="bottom" width="300" class="tf-value-container">
      <div slot-key="default">
        <mock-el-select value="Positive">
          <mock-el-option label="Positive" value="Positive"></mock-el-option>
          <mock-el-option label="Negative" value="Negative"></mock-el-option>
        </mock-el-select>
      </div>
      <div slot-key="reference">
        <div class="tf-value"><span class="tf-value-span"><span>Positive</span></span>
        </div>
      </div>
    </mock-el-popover>
    <div class="tf-remove el-icon-circle-close"></div>
  </div>
  <div class="tf-outer" data-test-key="project" options="">
    <div class="tf-name">
      Project:
    </div>
    <mock-el-popover trigger="click" placement="bottom" width="300" class="tf-value-container">
      <div slot-key="default">
        <mock-el-select placeholder="Start typing name" remote="" filterable="" clearable="" remote-method="function () { [native code] }" loading="true" loading-text="Loading matching entries..." no-match-text="No matches" multiple-limit="10" value="4567">
          <mock-el-option value="4567"></mock-el-option>
        </mock-el-select>
      </div>
      <div slot-key="reference">
        <div class="tf-value"><span class="tf-value-span"><span>
      
    </span>
          <!---->
          <!---->
          </span>
        </div>
      </div>
    </mock-el-popover>
    <div class="tf-remove el-icon-circle-close"></div>
  </div>
  <div class="tf-outer" data-test-key="submitter" options="">
    <div class="tf-name">
      Submitter:
    </div>
    <mock-el-popover trigger="click" placement="bottom" width="300" class="tf-value-container">
      <div slot-key="default">
        <mock-el-select placeholder="Start typing name" remote="" filterable="" clearable="" remote-method="function () { [native code] }" loading="true" loading-text="Loading matching entries..." no-match-text="No matches" multiple-limit="10" value="89AB">
          <mock-el-option value="89AB"></mock-el-option>
        </mock-el-select>
      </div>
      <div slot-key="reference">
        <div class="tf-value"><span class="tf-value-span"><span>
      
    </span>
          <!---->
          <!---->
          </span>
        </div>
      </div>
    </mock-el-popover>
    <div class="tf-remove el-icon-circle-close"></div>
  </div>
  <div class="tf-outer" data-test-key="datasetIds" options="">
    <div class="tf-name">
      Dataset:
    </div>
    <mock-el-popover trigger="click" placement="bottom" width="900" class="tf-value-container">
      <div slot-key="default">
        <mock-el-select placeholder="Start typing name" remote="" filterable="" clearable="" remote-method="function () { [native code] }" loading="true" loading-text="Loading matching entries..." no-match-text="No matches" multiple="true" multiple-limit="10"
          value="CDEF,GHIJ">
          <mock-el-option value="CDEF"></mock-el-option>
          <mock-el-option value="GHIJ"></mock-el-option>
        </mock-el-select>
      </div>
      <div slot-key="reference">
        <div class="tf-value"><span class="tf-value-span"><!----> <span>
      (2 items selected)
    </span>
          <!---->
          </span>
        </div>
      </div>
    </mock-el-popover>
    <div class="tf-remove el-icon-circle-close"></div>
  </div>
  <div class="tf-outer" data-test-key="minMSM" filterkey="minMSM" options="">
    <div class="tf-name">
      Min. MSM:
    </div>
    <div class="tf-value"><span class="tf-value-span"><span>
      0.5
    </span></span>
    </div>
    <div class="tf-remove el-icon-circle-close"></div>
  </div>
  <div class="tf-outer" data-test-key="compoundName" filterkey="compoundName" options="">
    <div class="tf-name">
      Molecule:
    </div>
    <div class="tf-value"><span class="tf-value-span"><span>
      C8H20NO6P
    </span></span>
    </div>
    <div class="tf-remove el-icon-circle-close"></div>
  </div>
  <div class="tf-outer" data-test-key="adduct" filterkey="adduct">
    <div class="tf-name">
      Adduct:
    </div>
    <mock-el-popover trigger="click" placement="bottom" width="300" class="tf-value-container">
      <div slot-key="default">
        <mock-el-select filterable="true" value="+K">
          <mock-el-option label="[Madducts.<sub>0</sub>.adduct]⁺" value="adducts.0.adduct"></mock-el-option>
          <mock-el-option label="[Madducts.<sub>1</sub>.adduct]⁺" value="adducts.1.adduct"></mock-el-option>
        </mock-el-select>
      </div>
      <div slot-key="reference">
        <div class="tf-value"><span class="tf-value-span"><span>[M + K]undefined</span></span>
        </div>
      </div>
    </mock-el-popover>
    <div class="tf-remove el-icon-circle-close"></div>
  </div>
  <div class="tf-outer" data-test-key="mz" filterkey="mz" options="">
    <div class="tf-name">
      m/z:
    </div>
    <div class="tf-value"><span class="tf-value-span"><span>
      296.0659
    </span></span>
    </div> <span class="tf-value-suffix">± 0.0001</span>
    <div class="tf-remove el-icon-circle-close"></div>
  </div>
  <div class="tf-outer" data-test-key="fdrLevel" filterkey="fdrLevel">
    <div class="tf-name">
      FDR:
    </div>
    <mock-el-popover trigger="click" placement="bottom" width="300" class="tf-value-container">
      <div slot-key="default">
        <mock-el-select value="0.1">
          <mock-el-option label="5%" value="0.05"></mock-el-option>
          <mock-el-option label="10%" value="0.1"></mock-el-option>
          <mock-el-option label="20%" value="0.2"></mock-el-option>
          <mock-el-option label="50%" value="0.5"></mock-el-option>
        </mock-el-select>
      </div>
      <div slot-key="reference">
        <div class="tf-value"><span class="tf-value-span"><span>10%</span></span>
        </div>
      </div>
    </mock-el-popover>
    <!---->
  </div>
  <div class="tf-outer" data-test-key="group" options="">
    <div class="tf-name">
      Group:
    </div>
    <mock-el-popover trigger="click" placement="bottom" width="300" class="tf-value-container">
      <div slot-key="default">
        <mock-el-select placeholder="Start typing name" remote="" filterable="" clearable="" remote-method="function () { [native code] }" loading="true" loading-text="Loading matching entries..." no-match-text="No matches" multiple-limit="10" value="0123">
          <mock-el-option value="0123"></mock-el-option>
        </mock-el-select>
      </div>
      <div slot-key="reference">
        <div class="tf-value"><span class="tf-value-span"><span>
      
    </span>
          <!---->
          <!---->
          </span>
        </div>
      </div>
    </mock-el-popover>
    <div class="tf-remove el-icon-circle-close"></div>
  </div>
  <div class="tf-outer" data-test-key="organism" filterkey="organism">
    <div class="tf-name">
      Organism:
    </div>
    <mock-el-popover trigger="click" placement="bottom" width="300" class="tf-value-container">
      <div slot-key="default">
        <mock-el-select filterable="true" value="cow">
          <mock-el-option label="organisms" value="organisms"></mock-el-option>
          <mock-el-option label="organisms" value="organisms"></mock-el-option>
        </mock-el-select>
      </div>
      <div slot-key="reference">
        <div class="tf-value"><span class="tf-value-span"><span>cow</span></span>
        </div>
      </div>
    </mock-el-popover>
    <div class="tf-remove el-icon-circle-close"></div>
  </div>
  <div class="tf-outer" data-test-key="organismPart" filterkey="organismPart">
    <div class="tf-name">
      Organism part:
    </div>
    <mock-el-popover trigger="click" placement="bottom" width="300" class="tf-value-container">
      <div slot-key="default">
        <mock-el-select filterable="true" value="stomach">
          <mock-el-option label="organismParts" value="organismParts"></mock-el-option>
          <mock-el-option label="organismParts" value="organismParts"></mock-el-option>
        </mock-el-select>
      </div>
      <div slot-key="reference">
        <div class="tf-value"><span class="tf-value-span"><span>stomach</span></span>
        </div>
      </div>
    </mock-el-popover>
    <div class="tf-remove el-icon-circle-close"></div>
  </div>
  <div class="tf-outer" data-test-key="condition" filterkey="condition">
    <div class="tf-name">
      Organism condition:
    </div>
    <mock-el-popover trigger="click" placement="bottom" width="300" class="tf-value-container">
      <div slot-key="default">
        <mock-el-select filterable="true" value="hungry">
          <mock-el-option label="conditions" value="conditions"></mock-el-option>
          <mock-el-option label="conditions" value="conditions"></mock-el-option>
        </mock-el-select>
      </div>
      <div slot-key="reference">
        <div class="tf-value"><span class="tf-value-span"><span>hungry</span></span>
        </div>
      </div>
    </mock-el-popover>
    <div class="tf-remove el-icon-circle-close"></div>
  </div>
  <div class="tf-outer" data-test-key="growthConditions" filterkey="growthConditions">
    <div class="tf-name">
      Sample growth conditions:
    </div>
    <mock-el-popover trigger="click" placement="bottom" width="300" class="tf-value-container">
      <div slot-key="default">
        <mock-el-select filterable="true" value="paddock">
          <mock-el-option label="growthConditions" value="growthConditions"></mock-el-option>
          <mock-el-option label="growthConditions" value="growthConditions"></mock-el-option>
        </mock-el-select>
      </div>
      <div slot-key="reference">
        <div class="tf-value"><span class="tf-value-span"><span>paddock</span></span>
        </div>
      </div>
    </mock-el-popover>
    <div class="tf-remove el-icon-circle-close"></div>
  </div>
  <div class="tf-outer" data-test-key="ionisationSource" filterkey="ionisationSource">
    <div class="tf-name">
      Source:
    </div>
    <mock-el-popover trigger="click" placement="bottom" width="300" class="tf-value-container">
      <div slot-key="default">
        <mock-el-select filterable="true" value="MALDI">
          <mock-el-option label="ionisationSources" value="ionisationSources"></mock-el-option>
          <mock-el-option label="ionisationSources" value="ionisationSources"></mock-el-option>
        </mock-el-select>
      </div>
      <div slot-key="reference">
        <div class="tf-value"><span class="tf-value-span"><span>MALDI</span></span>
        </div>
      </div>
    </mock-el-popover>
    <div class="tf-remove el-icon-circle-close"></div>
  </div>
  <div class="tf-outer" data-test-key="maldiMatrix" filterkey="maldiMatrix">
    <div class="tf-name">
      Matrix:
    </div>
    <mock-el-popover trigger="click" placement="bottom" width="300" class="tf-value-container">
      <div slot-key="default">
        <mock-el-select filterable="true" value="DHB">
          <mock-el-option label="maldiMatrices" value="maldiMatrices"></mock-el-option>
          <mock-el-option label="maldiMatrices" value="maldiMatrices"></mock-el-option>
        </mock-el-select>
      </div>
      <div slot-key="reference">
        <div class="tf-value"><span class="tf-value-span"><span>DHB</span></span>
        </div>
      </div>
    </mock-el-popover>
    <div class="tf-remove el-icon-circle-close"></div>
  </div>
  <div class="tf-outer" data-test-key="analyzerType" filterkey="analyzerType">
    <div class="tf-name">
      Analyzer type:
    </div>
    <mock-el-popover trigger="click" placement="bottom" width="300" class="tf-value-container">
      <div slot-key="default">
        <mock-el-select filterable="true" value="FTICR">
          <mock-el-option label="analyzerTypes" value="analyzerTypes"></mock-el-option>
          <mock-el-option label="analyzerTypes" value="analyzerTypes"></mock-el-option>
        </mock-el-select>
      </div>
      <div slot-key="reference">
        <div class="tf-value"><span class="tf-value-span"><span>FTICR</span></span>
        </div>
      </div>
    </mock-el-popover>
    <div class="tf-remove el-icon-circle-close"></div>
  </div>
  <div class="tf-outer" data-test-key="database" filterkey="database">
    <div class="tf-name">
      Database:
    </div>
    <mock-el-popover trigger="click" placement="bottom" width="300" class="tf-value-container">
      <div slot-key="default">
        <mock-el-select filterable="true" value="HMDBv4">
          <mock-el-option label="molecularDatabases.0.name" value="molecularDatabases.0.name"></mock-el-option>
          <mock-el-option label="molecularDatabases.1.name" value="molecularDatabases.1.name"></mock-el-option>
        </mock-el-select>
      </div>
      <div slot-key="reference">
        <div class="tf-value"><span class="tf-value-span"><span>HMDBv4</span></span>
        </div>
      </div>
    </mock-el-popover>
    <!---->
  </div>
  <div class="tf-outer" data-test-key="metadataType" filterkey="metadataType">
    <div class="tf-name">
      Data type:
    </div>
    <mock-el-popover trigger="click" placement="bottom" width="300" class="tf-value-container">
      <div slot-key="default">
        <mock-el-select filterable="true" value="ims">
          <mock-el-option label="Imaging MS" value="Imaging MS"></mock-el-option>
          <mock-el-option label="LC-MS" value="LC-MS"></mock-el-option>
        </mock-el-select>
      </div>
      <div slot-key="reference">
        <div class="tf-value"><span class="tf-value-span"><span>ims</span></span>
        </div>
      </div>
    </mock-el-popover>
    <!---->
  </div>
</div>
`;

exports[`FilterPanel should match snapshot (no filters) 1`] = `
<div class="filter-panel">
  <mock-el-select placeholder="Add filter" class="filter-select">
    <mock-el-option value="database" label="Select database"></mock-el-option>
    <mock-el-option value="fdrLevel" label="Select FDR level"></mock-el-option>
    <mock-el-option value="group" label="Select group"></mock-el-option>
    <mock-el-option value="project" label="Select project"></mock-el-option>
    <mock-el-option value="submitter" label="Select submitter"></mock-el-option>
    <mock-el-option value="datasetIds" label="Select dataset"></mock-el-option>
    <mock-el-option value="compoundName" label="Search molecule"></mock-el-option>
    <mock-el-option value="mz" label="Search by m/z"></mock-el-option>
    <mock-el-option value="polarity" label="Select polarity"></mock-el-option>
    <mock-el-option value="adduct" label="Select adduct"></mock-el-option>
    <mock-el-option value="organism" label="Select organism"></mock-el-option>
    <mock-el-option value="organismPart" label="Select organism part"></mock-el-option>
    <mock-el-option value="condition" label="Select organism condition"></mock-el-option>
    <mock-el-option value="growthConditions" label="Select sample growth conditions"></mock-el-option>
    <mock-el-option value="analyzerType" label="Select analyzer"></mock-el-option>
    <mock-el-option value="ionisationSource" label="Select ionisation source"></mock-el-option>
    <mock-el-option value="maldiMatrix" label="Select MALDI matrix"></mock-el-option>
    <mock-el-option value="minMSM" label="Set minimum MSM score"></mock-el-option>
    <mock-el-option value="simpleQuery" label="Search anything"></mock-el-option>
    <mock-el-option value="metadataType" label="Select data type"></mock-el-option>
  </mock-el-select>
</div>
`;
